model/version.go

```diff
@@ -601,6 +605,98 @@ func GetVersionsWithOptions(projectName string, opts GetVersionsOptions) ([]Vers
 	return res, nil
 }
 
+// ConstructManifest will construct a manifest from the given project and version.
+func ConstructManifest(v *Version, proj *Project, projectRef *ProjectRef, settings *evergreen.Settings) (*manifest.Manifest, error) {
+	if len(proj.Modules) == 0 {
+		return nil, nil
+	}
+	newManifest := &manifest.Manifest{
+		Id:          v.Id,
+		Revision:    v.Revision,
+		ProjectName: v.Identifier,
+		Branch:      projectRef.Branch,
+		IsBase:      v.Requester == evergreen.RepotrackerVersionRequester,
+	}
+	token, err := settings.GetGithubOauthToken()
+	if err != nil {
+		return nil, errors.Wrap(err, "getting github oauth token")
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	var gitBranch *github.Branch
+	var gitCommit *github.RepositoryCommit
+	modules := map[string]*manifest.Module{}
+	for _, module := range proj.Modules {
+		var sha, url string
+		owner, repo := module.GetRepoOwnerAndName()
+		if module.Ref == "" {
+			var commit *github.RepositoryCommit
+			commit, err = thirdparty.GetCommitEvent(ctx, token, projectRef.Owner, projectRef.Repo, v.Revision)
+			if err != nil {
+				return nil, errors.Wrapf(err, "can't get commit '%s' on '%s/%s'", v.Revision, projectRef.Owner, projectRef.Repo)
+			}
+			if commit == nil || commit.Commit == nil || commit.Commit.Committer == nil {
+				return nil, errors.New("malformed GitHub commit response")
+			}
+			// If this is a mainline commit, retrieve the module's commit from the time of the mainline commit.
+			// Otherwise, retrieve the module's commit from the time of the patch creation.
+			revisionTime := time.Unix(0, 0)
+			if !evergreen.IsPatchRequester(v.Requester) {
+				revisionTime = commit.Commit.Committer.GetDate()
+			}
+			var branchCommits []*github.RepositoryCommit
+			branchCommits, _, err = thirdparty.GetGithubCommits(ctx, token, owner, repo, module.Branch, revisionTime, 0)
+			if err != nil {
+				return nil, errors.Wrapf(err, "retrieving getting git branch for module %s", module.Name)
+			}
+			if len(branchCommits) > 0 {
+				sha = branchCommits[0].GetSHA()
+				url = branchCommits[0].GetURL()
+			}
+		} else {
+			sha = module.Ref
+			gitCommit, err = thirdparty.GetCommitEvent(ctx, token, owner, repo, module.Ref)
+			if err != nil {
+				return nil, errors.Wrapf(err, "retrieving getting git commit for module %s with hash %s", module.Name, module.Ref)
+			}
+			if gitCommit != nil {
+				url = *gitCommit.URL
+			}
+		}
+
+		modules[module.Name] = &manifest.Module{
+			Branch:   module.Branch,
+			Revision: sha,
+			Repo:     repo,
+			Owner:    owner,
+			URL:      url,
+		}
+
+	}
+	newManifest.Modules = modules
+	grip.Debug(message.Fields{
```

Comment:
I'm inclined to remove this (we have a lot of debugging statements and it's unclear what they're useful for; if we've created the manifest we theoretically should be able to find it in the db to debug with)